business/istio_status.go

```diff
@@ -44,25 +45,35 @@ type IstioStatusService struct {
 	workloads           *WorkloadService
 }
 
-func (iss *IstioStatusService) GetStatus(ctx context.Context) (kubernetes.IstioComponentStatus, error) {
+func (iss *IstioStatusService) GetStatus(ctx context.Context, cluster string) (kubernetes.IstioComponentStatus, error) {
```

Comment:
Do we need to always pass `cluster`? This would mean that anything calling this, like the frontend, would first need the full list of clusters to pass in. For example, the `api/mesh` handler now seems like it's not working as it was before:
```
 // @ TODO multi-cluster support, loop over control planes
 gi.IstioStatusGetter.GetStatus(ctx, config.Get().KubernetesConfig.ClusterName)
```

Can we instead loop over the `iss.userClients` to get the cluster list inside of this method and always return the complete `kubernetes.IstioComponentStatus`?

Reply:
I kept the cluster param for a reason of not to change the REST API call format, where in most of the calls we provide clusterName param, also to keep the result JSON format of arrays of component statuses.

Reply:
It can still be an array of component status. Isn't `Cluster` a field of each component status? So we don't need to separate by cluster before returning to the frontend.